HDFC BANK DCF VALUATION MODEL

In [53]:
import numpy as np
import pandas as pd

In [54]:
fcf = 11690.3
growth_rate_1to5 = 0.07
growth_rate_6to10 = 0.06
terminal_growth = 0.04
discount_rate = 0.1

In [55]:
projected_fcf = []
fcf_current = fcf

for year in range(1, 11):
    if year <= 5:
        fcf_current = fcf_current * (1 + growth_rate_1to5)
    else:
        fcf_current = fcf_current * (1 + growth_rate_6to10)
    projected_fcf.append(fcf_current)

pv_list = []
for year, fcf_year in enumerate(projected_fcf,start = 1):
  pv = fcf_year / (1+discount_rate)**year
  pv_list.append(pv)

In [56]:
print(projected_fcf)

[12508.621, 13384.22447, 14321.1201829, 15323.598595703, 16396.250497402212, 17380.025527246347, 18422.82705888113, 19528.196682413996, 20699.888483358838, 21941.88179236037]


In [57]:
print(pv_list)

[11371.473636363635, 11061.34253719008, 10759.669558903077, 10466.224025478448, 10180.781552056309, 9810.571313799715, 9453.82326602518, 9110.047874533355, 8778.773406368506, 8459.54528250056]


In [58]:
terminal_value = projected_fcf[-1]*(1 + terminal_growth) / (discount_rate - terminal_growth)
print(f"The terminal value is {terminal_value} million dollars")


The terminal value is 380325.95106757974 million dollars


In [59]:
PV_terminal = terminal_value/(1 + discount_rate)**10
PV_terminal

146632.11823000968

In [60]:
df = pd.DataFrame({
    "year": [1,2,3,4,5,6,7,8,9,10],
    "projected_fcf": projected_fcf,
    "present_value": pv_list
})
df

,year,projected_fcf,present_value
0,1,12508.621000,11371.473636
1,2,13384.224470,11061.342537
2,3,14321.120183,10759.669559
3,4,15323.598596,10466.224025
4,5,16396.250497,10180.781552
5,6,17380.025527,9810.571314
6,7,18422.827059,9453.823266
7,8,19528.196682,9110.047875
8,9,20699.888483,8778.773406
9,10,21941.881792,8459.545283


In [61]:
enterprise_value = sum(pv_list) + PV_terminal
print(f"The enterprise value is {enterprise_value} million dollars")

The enterprise value is 246084.37068322854 million dollars


In [62]:
#Sensitivity Analysis

pv_list = []
for year,fcf_year in enumerate(projected_fcf, start = 1):
  pv = fcf_year / (1+discount_rate)**year
  pv_list.append(pv)

for discount_rate in [0.07,0.08,0.09,0.10,0.11]:
  terminal_value = projected_fcf[-1] * (1 + terminal_growth) / (discount_rate - terminal_growth)
  terminal_pv = terminal_value / (1 + discount_rate) ** 10
  ev = sum(pv_list) + terminal_pv
  print (f"Discount rate {discount_rate:.0%}: EV = {ev:,.0f} million dollars")

Discount rate 7%: EV = 486,129 million dollars
Discount rate 8%: EV = 363,699 million dollars
Discount rate 9%: EV = 292,237 million dollars
Discount rate 10%: EV = 246,084 million dollars
Discount rate 11%: EV = 214,262 million dollars


The actual market cap for HDFC Bank is 136.6 Billion Dollars (136600 million dollars)

By considering a discount rate of 10% and computing the enterprise value corresponding to that, we get the figure 246.084 Billion dollars (246084.37 million dollars) which means the company is undervalued

At a 10% discount rate, our DCF model estimates HDFC Bank's intrinsic value at $246.08 billion, compared to its actual market cap of $136.6 billion, suggesting the stock is undervalued by the market. However, the market may be pricing in risks that our model ignores — specifically the RBI regulations and incorrectness of using DCF as a metric to evaluate banks.Additionally, DCF may not be the most appropriate valuation tool for banks, as it assumes FCF as the primary value driver. Banks are better valued using the Price-to-Book (P/B) ratio, which accounts for how efficiently a bank uses its capital.